core/build.gradle

```diff
@@ -120,12 +89,7 @@ test {
 class ProGuardTask extends proguard.gradle.ProGuardTask {
     def keepclasses(String... classNames) {
         classNames.each {
-            keep "public class $it { *; }"
-        }
-    }
-    def keepinterfaces(String... interfaceNames) {
-        interfaceNames.each {
-            keep "public interface $it { *; }"
+            keep "class $it { *; }"
```

Comment:
I believe I had tried using class to preserve the interfaces and it didn't work so I created keepinterfaces.

I think it is working anyways since you are proguarding the armeria source as well, so proguard knows we want the non-boxed methods.